pkgs/fc/agent/fc/manage/manage.py

```diff
@@ -264,6 +274,95 @@ def notify(category):
         self.log.info("maintenance-register-succeeded")
 
 
+class SwitchFailed(Exception):
+    pass
+
+
+@dataclass
+class CheckResult:
+    errors: list[str]
+    warnings: list[str]
+
+    def format_output(self) -> str:
+        if self.errors:
+            return "CRITICAL: " + " ".join(self.errors + self.warnings)
+
+        if self.warnings:
+            return "WARNING: " + " ".join(self.warnings)
+
+        return "OK: no problems found."
+
+    @property
+    def exit_code(self) -> int:
+        if self.errors:
+            return 2
+
+        if self.warnings:
+            return 1
+
+        return 0
+
+
+def check(log, enc) -> CheckResult:
+    errors = []
+    warnings = []
+    if INITIAL_RUN_MARKER.exists():
+        warnings.append(
+            f"{INITIAL_RUN_MARKER} exists. Looks like the agent has not "
+            f"run successfully, yet."
+        )
+
+    if STATE_VERSION_FILE.exists():
+        state_version = STATE_VERSION_FILE.read_text().strip()
+        log.debug("check-state-version", state_version=state_version)
+        if not re.match(r"\d\d\.\d\d", state_version):
+            warnings.append(
+                f"State version invalid: {state_version}, should look like 22.05"
+            )
+    else:
+        warnings.append(f"State version file {STATE_VERSION_FILE} missing.")
+
+    # ENC data checks
+    enc_params = enc["parameters"]
+
+    environment_url = enc_params.get("environment_url")
+    production_flag = enc_params.get("production")
+
+    log.debug(
+        "check-enc",
+        environment_url=environment_url,
+        production_flag=production_flag,
+    )
+
+    if production_flag is None:
+        errors.append("ENC: production flag is missing.")
+
+    if environment_url is None:
+        errors.append("ENC: environment URL is missing.")
+
+    uses_local_checkout = (
+        environment_url.startswith("file://") if environment_url else None
+    )
+
+    if production_flag and uses_local_checkout:
+        warnings.append("production VM uses local dev checkout.")
+
+    # nixos channel checks (missing/malformed)
+    nixos_channel = nixos.current_nixos_channel_url(log)
+    if nixos_channel is None:
+        errors.append("`nixos` channel not set.")
+    elif not (
+        nixos.get_fc_channel_build(nixos_channel, log)
```

Comment:
Please add a comment explaining how these conditions signal a state where the channel is still allowed to be misformed. This combination of conditions is not necessarily obvious.